### ISSUE TITLE
False positive on removing appcompat/

### ISSUE THREAD

**Issue by ColtonIdle** [autonomousapps/dependency-analysis-android-gradle-plugin#445]
**Plugin version**
0.75.0

**Gradle version**
7.1.1

**(Optional) Android Gradle Plugin (AGP) version**
7.1.0-alpha05

**Describe the bug**
buildHealth told me to remove both of my dependencies as they were unneeded. When I did. I got an error when building
```
error: cannot find symbol
                public void handleError(Async<?> operation, @NonNull Throwable throwable) {
                                                             ^
```

**To Reproduce**
Build file is

```
plugins {
    id("com.android.library")
    id("kotlin-android")
}

android {
    compileSdk = 29

    defaultConfig {
        minSdk = 26
        targetSdk = 29
    }

    buildTypes {
        named("release") {
            isMinifyEnabled = false
            proguardFiles(
                getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
        }
    }
    compileOptions {
        sourceCompatibility = JavaVersion.VERSION_1_8
        targetCompatibility = JavaVersion.VERSION_1_8
    }
}

dependencies {
    implementation("androidx.appcompat:appcompat:1.4.0-alpha03")
    implementation("androidx.core:core-ktx:1.6.0")
}

repositories { mavenCentral() }

```

I removed both deps (I don't know which one actually contained the support annotation because I am on mobile right now) and it caused the issue. Adding them back in fixes it.

**Comment by autonomousapps** [comment]
Please send build scan or stacktrace, and plugin console output. 

On August 1, 2021, GitHub ***@***.***> wrote:
> Plugin version
>  0.75.0
>
> Gradle version
>  7.1.1
>
> (Optional) Android Gradle Plugin (AGP) version
>  7.1.0-alpha05
>
> Describe the bug
>  buildHealth told me to remove both of my dependencies as they were
> unneeded. When I did. I got an error when building
>
> error: cannot find symbol public void handleError(Async<?> operation,
> @NonNull Throwable throwable) { ^
> To Reproduce
>  Build file is
>
> plugins { id("com.android.library") id("kotlin-android") } android {
> compileSdk = 29 defaultConfig { minSdk = 26 targetSdk = 29 }
> buildTypes { named("release") { isMinifyEnabled = false proguardFiles(
> getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-
> rules.pro") } } compileOptions { sourceCompatibility =
> JavaVersion.VERSION_1_8 targetCompatibility = JavaVersion.VERSION_1_8
> } } dependencies { implementation("androidx.appcompat:appcompat:1.4.0-
> alpha03") implementation("androidx.core:core-ktx:1.6.0") }
> repositories { mavenCentral() }
> I removed both deps (I don't know which one actually contained the
> support annotation because I am on mobile right now) and it caused the
> issue. Adding them back in fixes it.
>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-
plugin/issues/445>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PLJXDV4PDPZ422I4KDT2XR7DANCNFSM5BLV2DCA>.

**Comment by autonomousapps** [comment]
The annotation `@NonNull` does not come from either of those dependencies. It comes from `androidx.annotation:annotation`, which is a transitive dependency of both of those that you removed. I suspect the plugin advice suggested you should add it in addition to removing the unused dependencies.

**Comment by ColtonIdle** [comment]
Sorry. In the midst of a job change so I will not get back to this for some time. Feel free to close if you must!

**Comment by manask88** [comment]
i am having a similar issue. its warning me on the appcompat library. when i remove it, i get this error
`error: cannot access AppCompatImageView
            _loadingSpinner = itemView.findViewById(R.id.loading_spinner);
`

it seems like the only way to get AppCompatImageView is through androidx.appcompat:appcompat

**Comment by autonomousapps** [comment]
@manask88 do you have a reproducer, or can you describe your exact setup and repro steps?

**Comment by autonomousapps** [comment]
Closing as stale.
